### ISSUE TITLE
Vanity.playground.experiment doesn't choose the right experiment in an integration test

### ISSUE THREAD

**Issue by kareem** [assaf/vanity#199]
While writing an integration test using capybara, I've added these two lines to select an experiment to test against:

```
Vanity.context = Struct.new(:vanity_identity).new('1')
Vanity.playground.experiment(:experiment_name).chooses("control")
```

The control is never shown.  I've tried showing the variation:

```
Vanity.context = Struct.new(:vanity_identity).new('1')
Vanity.playground.experiment(:experiment_name).chooses("variation_a")
```

But the same version is always shown (the variation).

Any ideas on what I'm missing here?  Thanks.

Edit: using rspec.

**Comment by mrageh** [comment]
Hey @phillbaker 

This issue may be related to the issue I've been facing today.

Using version 2.1.2 of Vanity:

Within our application I call `Vanity.ab_test(:name_of_experiment)` and this gives me back one of our alternatives and I've set `use_vanity` in my controller.

I've been trying to write a test that uses `Vanity.playground.experiment(:name_of_experiment).chooses(:alternative)`, however the alternative that I've set in `chooses` is sometimes not returned. 

After digging into the codebase it seems like [`AbTest#chooses`](https://github.com/assaf/vanity/blob/master/lib/vanity/experiment/ab_test.rb#L232) set the correct identity of the `alternative` to return on the `@showing` instance variable. But when we call [`Vanity::Helpers#ab_test`](https://github.com/assaf/vanity/blob/master/lib/vanity/helpers.rb#L36) internally it calls [`AbTest#choose`](https://github.com/assaf/vanity/blob/master/lib/vanity/experiment/ab_test.rb#L201) which ignores what we've set and recalculates the `identity` of the alternative. 

This sometimes causes my spec to fail and I was wondering if what I've described above is a potential bug?

**Comment by phillbaker** [comment]
Hi there! Sorry for the slow response on this, that does sound like a bug,
do you mind sharing a simplified version of the test that's failing?

**Comment by mrageh** [comment]
No problem @phillbaker, I'll share a simplified test case with you sometime this week

**Comment by phillbaker** [comment]
@mrageh I merged a fix to #305 which may have been involved in this. Mind giving master a try?
